### PR TITLE
feat(rust-sdk): public async APIs

### PR DESCRIPTION
**Pull request by agustinustheo**
### Changelog
- Wrote `async` versions of the APIs.
- Install `async_trait` to allow `async fn` in traits.

### CODE REVIEW

File: rust/src/main.rs [hunk 1]
```diff
@@ -3,7 +3,62 @@ mod private;
...
+    }
+    match public::get_ticker_async("btc_usdt_spot").await {
+        Err(e) => println!("{:?}", e),
+        _ => ()
+    }
+    match public::get_trades_async("2", "btc_usdt_spot").await {
+        Err(e) => println!("{:?}", e),
+        _ => ()
+    }
+    match public::get_depth_async("btc_usdt_spot", "5").await {
```
Comment (dendisuhubdy) [inline rust/src/main.rs]: don't hardcode the symbol, use the parameter instrument

**Review by dendisuhubdy — CHANGES_REQUESTED** [review]

File: rust/src/private.rs [hunk 2]
```diff
@@ -98,4 +98,37 @@ pub fn get_account_balance() -> Result<(), Box<dyn Error>> {
...
+        .timeout(Duration::from_secs(config::timeout()))
+        .build()?;
+    let temp = URLBuilder.format (
+        config::url_api_bitwyre(),
+        config::get_private_api_endpoint(&"ACCOUNT_BALANCE"),
+        &param
+    );
+    let res = client.get(&temp)
+        .header("API_Key", api_key)
+        .header("API_Sign", signature)
```
Comment (thomsoe84) [inline rust/src/private.rs]: this one should be "API-Key" & "API-Sign" otherwise it will produce a 422 error just like in my PR before this

Comment (agustinustheo) [reply rust/src/private.rs]: changed, please check again.

File: rust/src/main.rs (hunk 1 above)
Comment (agustinustheo) [reply rust/src/main.rs]: Sorry, I don't understand, can you please elaborate?

**Comment by agustinustheo** [comment]
To make the code review more digestible, I decided to separate the tests and private async APIs to a new PR instead. @dendisuhubdy @thomsoe84

File: rust/src/main.rs (hunk 1 above)
Comment (thomsoe84) [reply rust/src/main.rs]: > I think what mas dendi means is you put all the instrument symbols in config.rs just like in babingepet repo, so to use the symbol we don't harcode it directly. CMIIW mas den

File: rust/src/private.rs [hunk 3]
```diff
@@ -117,8 +117,8 @@ pub async fn get_account_balance_async() -> Result<(), Box<dyn Error>> {
         &param
     );
     let res = client.get(&temp)
-        .header("API_Key", api_key)
-        .header("API_Sign", signature)
+        .header("API-Key", api_key)
+        .header("API-Sign", signature)
```
Comment (thomsoe84) [inline rust/src/private.rs:324]: oke nice @agustinustheo

File: rust/src/main.rs (hunk 1 above)
Comment (agustinustheo) [reply rust/src/main.rs]: Oh I get it, this is what you meant right? https://github.com/bitwyre/sdk/blob/dbf08f47d71b8ff67e6ccc52d2e2119793120ca6/rust/src/main.rs#L16

Comment (agustinustheo) [reply rust/src/main.rs]: I resolved this, can you please re-review? @dendisuhubdy

**Review by dendisuhubdy — APPROVED** [review]

**Review by dendisuhubdy — APPROVED** [review]

**Comment by dendisuhubdy** [comment]
@agustinustheo hi theo you need to rebase on top of origin/develop again by doing

```
git fetch --all
git rebase origin/develop
git push -f origin branchname
```

but don't forget to resign commits afterwards. You can follow this tutorial https://superuser.com/a/1123928

or do `git rebase --exec 'git commit --amend --no-edit -n -S' -i origin/develop`

**Comment by thomsoe84** [comment]
Approved

File: rust/src/main.rs [hunk 4]
```diff
@@ -4,8 +4,106 @@ mod private;
 use std::error::Error;
 use crate::public::config;
 
+async fn async_main() {
+    let instrument = config::bitwyre_instrument("1");
+    let instrument_contract = config::bitwyre_instrument("7");
```
Comment (thomsoe84) [inline rust/src/main.rs:9]: 👍

**Review by thomsoe84 — APPROVED** [review]

**Pull request merged**